calico_kubernetes/calico_kubernetes.py

```diff
@@ -709,18 +709,24 @@ def _log_interfaces(namespace):
     :param namespace
     :type namespace str
     """
-    if logger.isEnabledFor(logging.DEBUG):
-        interfaces = check_output(['ip', 'addr'])
-        logger.debug("Interfaces in default namespace:\n%s", interfaces)
-
-        namespaces = check_output(['ip', 'netns', 'list'])
-        logger.debug("Namespaces:\n%s", namespaces)
-
-        namespace_interfaces = check_output(['ip', 'netns',
-                                             'exec', str(namespace),
-                                             'ip', 'addr'])
-        logger.debug("Interfaces in namespace %s:\n%s",
-                     namespace, namespace_interfaces)
+    try:
+        if logger.isEnabledFor(logging.DEBUG):
+            interfaces = check_output(['ip', 'addr'])
+            logger.debug("Interfaces in default namespace:\n%s", interfaces)
+
+            namespaces = check_output(['ip', 'netns', 'list'])
+            logger.debug("Namespaces:\n%s", namespaces)
+
+            # Use nsenter because it doesn't rely on
+            # symlinks in /var/run/netns.
+            cmd = ['nsenter', '-t', str(namespace), '-n', 'ip', 'addr']
```

Comment:
`nsenter` isn't installed on all systems. I'd suggest sticking with the `ip netns` command here, though you could make a check to see if `/var/run/netns/<ns>` exists first, and print a friendly warning in that case.

Definitely good to pokemon-catch here as well though; we shouldn't ever fail in this branch.